core/common/src/MapLayerSettings.ts

```diff
@@ -135,11 +130,27 @@ export interface MapLayerProps {
   transparentBackground?: boolean;
   /** Is a base layer.  Defaults to 'false'. */
   isBase?: boolean;
+
+}
+
+export interface ImageMapLayerProps extends MapLayerPropsBase {
+  /** URL */
+  url: string;
+  /** Identifies the map layers source.*/
+  formatId: string;
+  /** Source layers. If undefined all layers are displayed. */
+  subLayers?: MapSubLayerProps[];
+
   /** Access Key for the Layer, like a subscription key or access token.
    * TODO This does not belong in the props object. It should never be persisted.
    */
   accessKey?: MapLayerKey;
 }
+export interface ModelMapLayerProps extends MapLayerPropsBase {
+  modelId: Id64String;
+}
+
+export type MapLayerProps = ImageMapLayerProps | ModelMapLayerProps;
```

Comment:
This will lead to a lot of 'instanceof' check in the code, don't know we could do better though.  @pmconne ?

Reply:
Can't use `instanceof` on interface types. Can do things with properties to allow compiler to deduce type.

Reply:
...there are no instanceof checks for this.  It tests for specific properties (url or modelId) to create the correct type (ModelMapLayerSettings or ImageMapLayerSettings)